src/components/Datetime.tsx

```diff
@@ -0,0 +1,190 @@
+import { useState, useRef } from 'react';
+
+const Datetime = () => {
+  const [typeOfState, setTypeOfState] = useState<'success' | 'error'>('success');
+  const [typeOfSize, setTypeOfSize] = useState<'sm' | 'md' | 'lg'>('lg');
+  const [typeOfInput, setTypeOfInput] = useState<'datetime-local' | 'date' | 'time'>('datetime-local');
+  const tdsDatetimeRef = useRef<HTMLTdsDatetimeElement>(null);
+  const [noMinWidth, setNoMinWidth] = useState(false); // State for noMinWidth
+	const [isDatetimeDisabled, setIsDatetimeDisabled] = useState(false);
+
+  const handleNewState = (event: Event) => {
+    const newState: 'error' | 'success' = (event.target as HTMLTdsRadioButtonElement).value as 'error' | 'success';
+    setTypeOfState(newState);
+  };
+
+  const handleType = (event: Event) => {
+    const newTypeOfInput: 'datetime-local' | 'date' | 'time' = (event.target as HTMLTdsRadioButtonElement).value as 'datetime-local' | 'date' | 'time';
+    setTypeOfInput(newTypeOfInput);
+  };
+
+  const handleSize = (event: Event) => {
+    const newSize: 'sm' | 'md' | 'lg' = (event.target as HTMLTdsRadioButtonElement).value as 'sm' | 'md' | 'lg';
+    setTypeOfSize(newSize);
+  };
+
+	const handleToggle = (event: CustomEvent) => {
+		const { detail } = event as CustomEvent<{ checked: boolean }>;
+		setNoMinWidth(detail.checked);
+	};
+
+	const handleDisabledState = (event: CustomEvent) => {
+		const { checked } = event.detail;
+		setIsDatetimeDisabled(checked);
+	};
```

Comment:
```suggestion
	const handleDisabledState = (event: CustomEvent<{ checked: boolean }>) => {
		const { detail } = event;
		setIsDatetimeDisabled(detail.checked);
	};
```

To make it consistent with the event handler directly above. 👆

Reply:
Nice, I agree with the feedback!